corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -311,6 +311,14 @@ def __init__(self,
     def app_id(self):
         return self.app.get_id if self.app else None
 
+    def __repr__(self):
+        return "RestoreParams(sync_log_id='{}', version={}, app='{}', device_id='{}'".format(
```

Comment:
nit: rather than manually adding quotes around a string, you can tell format to repr it directly. That way any quotes or other special characters in the string will be escaped appropriately.

```py
return "RestoreParams(sync_log_id={!r} ...".format(self.sync_log_id, ...)
```